include/mbgl/shaders/mtl/shader_program.hpp

```diff
@@ -62,7 +62,7 @@ class ShaderProgram final : public gfx::ShaderProgramBase {
 
     MTLRenderPipelineStatePtr getRenderPipelineState(const gfx::Renderable&,
                                                      const MTLVertexDescriptorPtr&,
-                                                     const gfx::ColorMode& colorMode) const;
+                                                     const gfx::ColorMode& colorMode);
```

Comment:
This would be a good use of the `mutable` keyword. You can use it on the (internal) cache to keep the method `const`.